### DISCUSSION TITLE
Load and Unload Filament

### DISCUSSION THREAD

**Discussion by jgbrown54** [Guizz27/DWIN_LK4Pro#11] [Ideas]
I recomend adding Load and Unload functions to the Filament screen. Basically an M701 and M702.

**Comment by Guizz27** [comment]
This is one the first thing I will add

**Comment by rispeed** [comment]
Do you know where I can get the stock UI firmware source ? I've been looking everywhere. I've already compiled marlin but cannot find the ui firmware anywhere except your modified one.

**Comment by Guizz27** [comment]
You can find a version here https://github.com/LONGER3D/Alfawise_U30_Pro_Screen_FW
This is an old version. I don't know if they modified the screen FW when they did a Marlin 2.0 version, but this should give you a good start. Look in the Alfawise_U30_Pro_Screen_Project for the DGUS DWIN project. This FW will only work if your screen FW has the letters DW or no letters in the version number. If your version number contains TL5, DWJT or DWJTB this FW will not work and I am not aware of source project for those screen versions.

**Reply by rispeed** [reply]
thank you much appreciated :)
